docarray/base_doc/doc.py

```diff
@@ -78,35 +94,144 @@ class MyDoc(BaseDoc):
         example=os.urandom(16).hex(),
     )
 
-    class Config:
-        json_loads = orjson.loads
-        json_dumps = orjson_dumps_and_decode
-        # `DocArrayResponse` is able to handle tensors by itself.
-        # Therefore, we stop FastAPI from doing any transformations
-        # on tensors by setting an identity function as a custom encoder.
-        json_encoders = {AbstractTensor: lambda x: x}
+    if is_pydantic_v2:
+
+        class Config:
```

Comment:
Should we still have this Config inner class, it is not recommended anymore in v2, right?
Also, where did the orjson stuff and the extra protobuf setting go?

Reply:
good question,

basically now in pydantic v2 this way is "deprecated". But at the same time, it is still working. Therefore if we keep it as a class it means we don't have to repeat the config for pydantic v2. Might be easier to maintain ...

Reply:
> Also, where did the orjson stuff and the extra protobuf setting go?

This is handle differently with pydantic v2 and therefore not needed

Reply:
hmm, using deprecated stuff is not great. Does it raise a warning?

Reply:
it does indeed raise a warning ...